### PR TITLE
feat(log): Move to `github.com/phuslu/log`

### PR DESCRIPTION
**Pull request by itsdevbear**
## Summary by CodeRabbit

- **New Features**
  - Introduced a custom log formatter enhancing log message clarity and format.
  - Added benchmarking for logging operations to evaluate performance.

- **Dependencies**
  - Added and updated multiple dependencies for improved performance and functionality.
  
- **Bug Fixes**
  - Updated several dependencies to their newer versions, improving stability and security.

### CODE REVIEW

File: mod/log/pkg/phuslu/phuslu.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+func (l *Logger[KeyValT, ImplT]) Impl() any {
+	return l.logger
+}
+
+func (l *Logger[KeyValT, ImplT]) With(keyVals ...KeyValT) ImplT {
+	newLogger := *l
+	r := make(map[string]interface{})
+	for _, keyVal := range keyVals {
+		r[keyVal.Key()] = keyVal.Value()
```
Comment (ocnc2) [inline mod/log/pkg/phuslu/phuslu.go:44]: not sure if we can do this if KeyValT is any

File: mod/node-core/pkg/builder/builder.go [hunk 2]
```diff
@@ -155,6 +160,7 @@ func (nb *NodeBuilder[NodeT]) buildRootCmd() (*cobra.Command, error) {
 		&autoCliOpts,
 		&mm,
 		&clientCtx,
+		&logger,
```
Comment (ocnc2) [inline mod/node-core/pkg/builder/builder.go:163]: provide logger

File: mod/log/pkg/phuslu/formatter.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+// customFormatter is a custom Formatter to pass into the ConsoleWriter.
+func customFormatter(out io.Writer, args *log.FormatterArgs) (int, error) {
+	b, ok := byteBufferPool.Get().(*byteBuffer)
+	if !ok {
+		panic("failed to get byte buffer from pool")
+	}
+	b.Bytes = b.Bytes[:0]
+	defer byteBufferPool.Put(b)
+
+	// TODO: pull out of config
```
[the remark this answers is not in the document]
Comment (ocnc2) [reply mod/log/pkg/phuslu/formatter.go:75]: after

File: mod/log/pkg/phuslu/phuslu_benchmarks_test.go [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+	"github.com/cosmos/cosmos-sdk/server"
+)
+
+// Benchmark function for phuslu logger.
+func BenchmarkPhusluLoggerInfo(b *testing.B) {
+	logger := phuslu.NewLogger[log.Logger]("Info", &bytes.Buffer{})
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		logger.Info("This is an info message", "key1", "value1", "key2", 2)
+	}
```
[the remark this answers is not in the document]
Comment (ocnc2) [reply mod/log/pkg/phuslu/phuslu_benchmarks_test.go:38]: newlogger ain't return no err

**Review by ocnc2 — COMMENTED** [review]
will optimize adapter + increase modularity in separate pr

File: mod/cli/go.mod [hunk 5]
```diff
@@ -32,6 +33,11 @@ require (
 	golang.org/x/sync v0.7.0
 )
 
+require (
+	github.com/phuslu/log v1.0.106 // indirect
+	github.com/rs/zerolog v1.33.0 // indirect
```
Comment (itsdevbear) [inline mod/cli/go.mod:38]: tbh comment out the sdk benchmark tests to prevent the zerolog dependency

File: mod/log/go.mod [hunk 6]
```diff
@@ -1,3 +1,155 @@
 module github.com/berachain/beacon-kit/mod/log
 
 go 1.22.4
+
+require (
+	cosmossdk.io/log v1.3.1
```
Comment (itsdevbear) [inline mod/log/go.mod:6]: these too, we shouldn't need these deps, bad to couple to this package

File: mod/log/pkg/phuslu/formatter.go [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+			b.Bytes = strconv.AppendQuote(b.Bytes, kv.Value)
+		} else {
+			fmt.Fprintf(b, " %s=%s", kv.Key, kv.Value)
+		}
+	}
+	// message
+	if endWithMessage {
+		fmt.Fprintf(b, " %s", args.Message)
+	}
+}
```
[the remark this answers is not in the document]
Comment (ocnc2) [reply mod/log/pkg/phuslu/formatter.go]: clean up later

**Pull request merged**